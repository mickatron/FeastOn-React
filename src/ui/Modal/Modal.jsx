import classnames from 'classnames';
import PropTypes from 'prop-types';
import React from 'react';
import ReactDOM from 'reactDOM';

import classNameType from 'js/shared/proptypes/className';

const modalRoot = document.getElementById('modal-root');

export default class Modal extends React.PureComponent {

  static propTypes = {
    className: classNameType,
    children: PropTypes.node.isRequired,
    open: PropTypes.bool,
  }

  static defautProps = {
    open: false,
  }

  $modal = undefined

  componentDidMount() {
    this.$modal = document.createElement('div');
    modalRoot.appendChild(this.$modal);
  }

  componentWillUnmount() {
    modalRoot.removeChild(this.$modal);
  }

  render() {
    const { className, children, open } = this.props;

    return open && ReactDOM.createPortal(
      <div 
        className={classnames(className, 'c-Modal')}
        ref={$modal => this.$modal=$modal}
      >
        {children}
      </div>,
      modalRoot,
    );
  }
}