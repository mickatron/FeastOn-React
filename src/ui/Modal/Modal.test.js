import { shallow } from 'enzyme';
import React from 'react';
import Modal from './Modal';

describe('Modal', () => {

  let component = null;

  beforeAll(() => {
    component = shallow(
      <Modal 
        className="test"
        open={true}
      >
        <p>Modal Content</p>
      </Modal>
    );
    expect(component).toMatchSnapshot();
  });

  describe('renders when open={true}', ()=>{
    it('has custom className', () => {    
      expect(component.find('.c-Modal.test').exists()).toBeTruthy();
    });
    
    it('has children', () => { 
      expect(component.find('.c-Modal p').exists()).toBeTruthy();
    });
  });

  it('does not render when open={false}', () => {
    component.setProps({open:false});
    expect(component.find('.c-Modal').exists()).toBeFalsy();
  });
  
});