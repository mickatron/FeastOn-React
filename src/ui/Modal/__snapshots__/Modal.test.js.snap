// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<div
  className="test c-Modal"
>
  <p>
    Modal Content
  </p>
</div>
`;
